src/tictactoe/Board.java

```diff
@@ -1,21 +1,116 @@
 package tictactoe;
 
+import java.util.*;
+
 public class Board {
     public Space[][] spaces;
-    private int size;
+    private int size, boardSize;
     public Board(int size) throws InvalidBoardSizeException {
         checkIfValidBoardSize(size);
         initializeSpaces();
+        boardSize = size * size;
+    }
+
+    public void fillSpace(Space space, Marker marker) throws SpaceDoesNotExistException {
+        Space spaceToFill = getSpace(space);
+        List<Space> spaces = availableSpaces();
+        if (validSpace(spaceToFill)) {
+            spaces.get(spaces.indexOf(spaceToFill)).fill(marker);
+        }
+    }
+
+    public void resetSpace(Space space) throws SpaceDoesNotExistException {
+        Space spaceToReset = getSpace(space);
+        List<Space> spaces = unavailableSpaces();
+        spaces.get(spaces.indexOf(spaceToReset)).reset();
+    }
+
+    public boolean isFull() {
+        List<Space> spaces = availableSpaces();
+        if (spaces.size() == 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public boolean isEmpty() {
+        List<Space> spaces = availableSpaces();
+        if (spaces.size() == boardSize) {
```

Comment:
Instead of using an `if else` block here just `return spaces.size() == boardSize;`